File: test/arguments/set_top_level_package/python/SetTopLevelPackageTest.py
import unittest

from testutils import getZserioApi


class SetTopLevelPackageTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(
            __file__,
            "set_top_level_package.zs",
            extraArgs=["-setTopLevelPackage", "company.appl"],
            topLevelPackage="company",
        )

    def testEmptyConstructor(self):
        simpleStructure = self.api.appl.SimpleStructure()
        simpleStructure.simple_choice = self.api.appl.SimpleChoice(simpleStructure.value)
        simpleStructure.simple_template = self.api.appl.SimpleTemplate_Enumeration(True, 0)
        self.assertEqual(32, simpleStructure.bitsizeof())

File: test/arguments/set_top_level_package/cpp/SetTopLevelPackageTest.cpp
#include "gtest/gtest.h"

#include "company/appl/SimpleStructure.h"

namespace company
{
namespace appl
{
namespace simple_structure
{

class SetTopLevelPackageTest : public ::testing::Test
{
protected:
    static const size_t SIMPLE_STRUCTURE_BIT_SIZE;
};

const size_t SetTopLevelPackageTest::SIMPLE_STRUCTURE_BIT_SIZE = 32;

TEST_F(SetTopLevelPackageTest, emptyConstructor)
{
    {
        SimpleStructure simpleStructure;
        simpleStructure.initializeChildren();
        ASSERT_EQ(SIMPLE_STRUCTURE_BIT_SIZE, simpleStructure.bitSizeOf());
    }

    {
        SimpleStructure simpleStructure = {};
        simpleStructure.initializeChildren();
        ASSERT_EQ(SIMPLE_STRUCTURE_BIT_SIZE, simpleStructure.bitSizeOf());
    }
}

TEST_F(SetTopLevelPackageTest, fieldConstructor)
{
    SimpleStructure simpleStructure2 = SimpleStructure({}, {}, {}, {}, {}, {});
    simpleStructure2.initializeChildren();
    ASSERT_EQ(SIMPLE_STRUCTURE_BIT_SIZE, simpleStructure2.bitSizeOf());
}

} // namespace simple_structure
} // namespace appl
} // namespace company
